sdk/core/Azure.Core.Experimental/src/BinaryDataExtensions.cs

```diff
@@ -16,7 +16,15 @@ public static class BinaryDataExtensions
         /// </summary>
         public static dynamic ToDynamic(this BinaryData data)
         {
-            return new DynamicJson(MutableJsonDocument.Parse(data).RootElement);
+            return new DynamicJson(MutableJsonDocument.Parse(data).RootElement, new DynamicJsonOptions());
```

Comment:
Default behavior is that you can get/set properties PascalCase (standard .NET style) and new properties are written in the JSON buffer camelCase (Azure API guidelines).

Reply:
I think we at least need an option parameter to JsonData ctor and to ToDynamic method to control this behavior. Something like:

```csharp
public enum JsonDataNameMapping {
    None,
    PascalCaseGetters,
    PascalCaseGettersCamelCaseSetters // default
}
```

Reply:
Does [this method](https://github.com/Azure/azure-sdk-for-net/pull/34082/files#diff-e99fb95729734f4c8b175ed7e726b01d65cf7a34efb5d2a500a2829c357fdab2R25) (just below this comment) address your suggestion?  

Or are you recommending replacing DynamicJsonOptions with the enum you've proposed?  Or a third `ToDynamic()` that takes the simplified `DynamicJsonNameMapping` enum?

Reply:
(`JsonData` has been replaced by `DynamicJson` in the new set of types.)

Reply:
Ah, sorry. I did not notice the method below. Thanks!